ov_core/src/track/TrackKLT.cpp

```diff
@@ -207,20 +226,30 @@ void TrackKLT::feed_stereo(double timestamp, cv::Mat &img_leftin, cv::Mat &img_r
 
     // Lets track temporally
 #ifdef ILLIXR_INTEGRATION
-    std::thread t_ll = timed_thread("slam2 matching l", &TrackKLT::perform_matching, this, boost::cref(img_pyramid_last[cam_id_left]), boost::cref(imgpyr_left),
-                                       boost::ref(pts_last[cam_id_left]), boost::ref(pts_left_new), cam_id_left, cam_id_left, boost::ref(mask_ll));
-    std::thread t_rr = timed_thread("slam2 matching r", &TrackKLT::perform_matching, this, boost::cref(img_pyramid_last[cam_id_right]), boost::cref(imgpyr_right),
-                                       boost::ref(pts_last[cam_id_right]), boost::ref(pts_right_new), cam_id_right, cam_id_right, boost::ref(mask_rr));
+	parallel_for_(cv::Range(0, 2), [&](const cv::Range& range){
+		for (int i = range.start; i < range.end; i++) {
```

Comment:
Out of curiosity, does this spawn 2 threads, or 3 threads (where the 3rd thread fails the `for` guard immediately?

Reply:
[`cv::Range`](https://docs.opencv.org/master/da/d35/classcv_1_1Range.html) is inclusive of the lower-bound and exclusive of the upper-bound, so the range contains 0 and 1.